src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs

```diff
@@ -23,10 +23,10 @@ public static class AWSLambdaWrapper
     private static readonly AssemblyName AssemblyName = typeof(AWSLambdaWrapper).Assembly.GetName();
 
     [SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1202:ElementsMustBeOrderedByAccess", Justification = "Initialization order.")]
-    internal static readonly string ActivitySourceName = AssemblyName.Name;
+    internal static readonly string? ActivitySourceName = AssemblyName.Name;
 
-    private static readonly Version Version = AssemblyName.Version;
-    private static readonly ActivitySource AWSLambdaActivitySource = new(ActivitySourceName, Version.ToString());
+    private static readonly Version? Version = AssemblyName.Version;
+    private static readonly ActivitySource AWSLambdaActivitySource = new(ActivitySourceName ?? "OpenTelemetry.Instrumentation.AWSLambda", Version?.ToString());
```

Comment:
Open question on what to do if `Version` is `null`: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1544#discussion_r1461645540

Reply:
What do other components do here?
If AoT does not provide this version info, should we auto-generate a C# file with msbuild during build to hold the version info?

Reply:
I don't know - I'll have to look into whether any other of the libraries here have already hit the same thing.

Reply:
Ignoring tests, the libraries fall into two camps:

1. Don't specify a version at all.
2. Pass through the assembly version, which won't work for native AoT as noted.

![image](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/assets/1439341/18cfe4a3-36b2-4652-96e1-3f1fe0d1e906)

I can test it out separately, but I think if the version wants to be kept, the better solution is to investigate whether native AoT retains [`AssemblyFileVersion]`/ `[AssemblyInformationalVersion]` to use instead.

This is already present in the assembly according to [nuget.info](https://nuget.info/packages/OpenTelemetry.Instrumentation.AWSLambda/1.2.0-beta.1):

![image](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/assets/1439341/e5921299-7099-402b-9295-dd0ee2ca05e9)